hail/python/hailtop/batch/backend.py

```diff
@@ -405,29 +408,26 @@ def __init__(self,
             warnings.warn('Use of deprecated positional argument \'bucket\' in ServiceBackend(). Specify \'bucket\' as a keyword argument instead.')
             bucket = args[1]
 
-        if billing_project is None:
-            billing_project = get_user_config().get('batch', 'billing_project', fallback=None)
+        billing_project = configuration_of('batch', 'billing_project', billing_project, None)
         if billing_project is None:
             raise ValueError(
                 'the billing_project parameter of ServiceBackend must be set '
                 'or run `hailctl config set batch/billing_project '
                 'MY_BILLING_PROJECT`')
         self._batch_client = BatchClient(billing_project, _token=token)
 
-        user_config = get_user_config()
-
         if bucket is not None:
             warnings.warn('Use of deprecated argument \'bucket\' in ServiceBackend(). Specify \'remote_tmpdir\' as a keyword argument instead.')
 
         if remote_tmpdir is not None and bucket is not None:
             raise ValueError('Cannot specify both \'remote_tmpdir\' and \'bucket\' in ServiceBackend(). Specify \'remote_tmpdir\' as a keyword argument instead.')
 
         if bucket is None and remote_tmpdir is None:
-            remote_tmpdir = user_config.get('batch', 'remote_tmpdir', fallback=None)
+            remote_tmpdir = configuration_of('batch', 'remote_tmpdir', remote_tmpdir, None)
 
         if remote_tmpdir is None:
             if bucket is None:
-                bucket = user_config.get('batch', 'bucket', fallback=None)
+                bucket = configuration_of('batch', 'bucket', bucket, None)
```

Comment:
Let's not make this change, it allows people to use the envvar HAIL_BATCH_BUCKET, which wasn't previously possible. I want to discourage people from specifying a bucket.